python/simpler/worker.py

```
@@ -455,24 +455,28 @@ class _ChildProvEntry:
     window / carved buffer pointer can legally alias the same device address.
-    The key is live while ``malloc_owned or domain_allocation_ids``; only an
-    exact ``malloc`` base is ``free``-able, while a domain pointer is revoked by
-    its domain's release. Interior pointers are never recorded, so a pointer
-    that merely lands inside a live allocation has no entry and is rejected.
+    Each role records its accessible extent from the exact base. Only a malloc
+    base is ``free``-able, while a domain pointer is revoked by its domain's
+    release. Interior pointers are never recorded as independent bases.
     """
 
-    __slots__ = ("malloc_owned", "domain_allocation_ids")
+    __slots__ = ("malloc_nbytes", "domain_allocation_nbytes")
```

Ruff (0.15.21)

[warning] `_ChildProvEntry.__slots__` is not sorted

Apply a natural sort to `_ChildProvEntry.__slots__`

(RUF023)